compiler/src/iree/compiler/Codegen/Common/VectorReductionToGPU.cpp

```diff
@@ -40,28 +42,77 @@ static Value allocateGlobalSharedMemory(Location loc, OpBuilder &builder,
   return builder.create<memref::AllocOp>(loc, memrefType);
 }
 
+/// Packs vector of lower precision into a single 32-bit width element.
+/// (i.e <2xf16> -> f32 and <4xi8> -> i32)
+static Value packVectorToSupportedWidth(Location loc, OpBuilder &builder,
+                                        Value input) {
+  auto vecType = input.getType().cast<VectorType>();
+  assert(vecType && "This packing fn only works if input is a vector.");
+  Type elementType = vecType.getElementType();
+  int64_t vecSize = vecType.getShape()[0];
+  unsigned bitWidth = elementType.getIntOrFloatBitWidth();
+  // Check that numel * bitwidth = 32.
+  assert(vecSize * bitWidth == kShuffleBitWidth &&
+         "vector needs to packable into 32-bitwidth.");
+  assert(elementType.isIntOrFloat() && "Only int and float supported.");
+  Type supportedType = builder.getF32Type();
+  if (elementType.dyn_cast<IntegerType>()) {
+    supportedType = builder.getI32Type();
```

Comment:
as discussed I would use i32 all the time

Reply:
done!